### ISSUE TITLE
Cannot setQueryParams on Template.XYZ.onCreated

### ISSUE THREAD

**Issue by OminStyle** [meteorhacks/flow-router#216]
http://meteorpad.com/pad/M88gpSPRcAvvn8iXt/Flow-router-default-query
Although Meteorpad doesn't seem to have the latest flow-router (I don't see @1.18.0 when I type @), I get the same behavior on my current project using v1.18.0. I ```setQueryParams``` when the template is created and the query param gets added and deleted in a split second. If I directly type in the url, the query param is there.

**Comment by arunoda** [comment]
I see. Let's see what we can do about this.
Enque all the tasks (may be using) Tracker.afterFlush before we do apply any changes.

**Comment by arunoda** [comment]
Hi,

I did half fix for this issue.
Now all the routing works pretty well.
But, you can't get the state just after set that. Getting state will be eventual consistence (big words)

But it'll work as reactively without any issues.

We can have a proper solution when we removed subscriptions. (on 3.0)

**Comment by OminStyle** [comment]
Although there is definitely some weird behavior going on, I'm starting to think that perhaps it's not meant to be used this way. I decided to remove setQueryParams on Template.XYZ.onCreated. I was initially planning to have a default query set in the url (eg. ```?date=today```), which I have removed, but if I were to add it again, I would do it from the routing layer, not the template layer.

Btw thanks for sharing all the awesome tools and tutorials. I'm using [Meteor UI Pattern: Keeping App State on the URL](https://meteorhacks.com/meteor-ui-pattern-keeping-app-state-in-the-url) in my current project.
